File: 0866-prime-palindrome/0866-prime-palindrome.py
class Solution:
    def primePalindrome(self, N):
        """
        :type N: int
        :rtype: int
        """
        ndigits = len(str(N))
        while True:
            for x in self.palindromes(ndigits):
                if x >= N and self.isPrime(x):
                    return x
            ndigits += 1
            
    def palindromes(self, n):
        if n == 1:
            for i in range(10):
                yield i
        elif n % 2 == 0:
            d = n // 2
            for i in range(10**(d-1), 10**d):
                s = str(i)
                yield int(s + s[::-1])
        else:
            d = n // 2
            for i in range(10**(d-1), 10**d):
                s = str(i)
                for j in range(10):
                    yield int(s + str(j) + s[::-1])
                    
    def isPrime(self, x):
        if x == 1:
            return False
        if x == 2:
            return True
        for i in range(2, int(x**0.5+1)):
            if x % i == 0:
                return False
        return True

File: 0866-prime-palindrome/0866-prime-palindrome.cpp
class Solution {
public:
    bool isPrime(int num) {
        if(num<2) 
            return false;
        if(num==2)
            return true;
        if(num%2==0)
            return false; 
        for(int i=3;i<=sqrt(num);i+=2){
            if(num%i==0)
                return false;
        }
        return true;
    }
    bool isPalindrome(int num) {
        string s =to_string(num);
        int left=0,right=s.size()-1;
        while(left<right){
            if(s[left++]!=s[right--])
                return false;
        }
        return true;
    }
    int primePalindrome(int n) {
        if(n>=8&&n<=11) 
            return 11;
        for(int num= n;;num++){
            if(isPalindrome(num)&&isPrime(num)) {
                return num;
            }
        }
    }
};


